packages/langium/src/lsp/document-update-handler.ts

```diff
@@ -141,4 +145,18 @@ export class DefaultDocumentUpdateHandler implements DocumentUpdateHandler {
             .toArray();
         this.fireDocumentUpdate(changedUris, deletedUris);
     }
+
+    didCloseDocument(event: TextDocumentChangeEvent<TextDocument>): void {
+        const document = this.documents.getDocument(URI.parse(event.document.uri));
+        if (document) {
+            // Preserve memory by discarding the CST of the document
+            // Whenever the user reopens the document, the CST will be rebuilt
+            discardCst(document.parseResult.value);
+        }
+        // Discard the diagnostics for the closed document
```

Comment:
Shall we make this configurable (without method overriding)? It could be simply a public property on the service class to turn this behavior on or off?

I imagine there are users/language implementors who would like to keep all problem markers even for closed documents.